### ISSUE TITLE
Wrong gradients if backpropagating through multiple forward calls

### ISSUE THREAD

**Issue by bauerfe** [synsense/sinabs-exodus#3]
There is a small bug in the way exodus calculates gradients. It only manifests if you try to backpropagate across multiple `forward` calls. For instance:
```
for inp0, inp1 in dataloader:
  out0 = snn(inp0)
  out1 = snn(inp1)
  loss = f(out0, out1)
  loss.backward()
  optimizer.step()
  snn.reset_states()
  snn.zero_grad()
```
will result in slightly different gradients than what would be expected in BPTT.

However, if you do the backward pass after each forward call and then reset the model (as is usually done), the bug does not occur and the gradients should be correct. E.g.:
```
for inp in dataloader:
  out = snn(inp)
  loss = f(out)
  loss.backward()
  optimizer.step()
  snn.reset_states()
  snn.zero_grad()
```
works fine.

I will try to fix this asap.
